app/src/main/java/com/pusher/demo/features/marketplace/ChatkitManager.kt

```diff
@@ -58,7 +73,25 @@ object ChatkitManager {
 
         //connect to chatkit
         chatManager.connect(
-            listeners = ChatListeners(),
+            listeners = ChatListeners(
+
+                onRoomUpdated = {
+                    if (chatListeners.size > 0) {
```

Comment:
Those `if` checks are redundant. It would be also better to store `ChatManagerEventConsumer`s (listener conversion: `toCallback`) and use the `connect` version that accepts the consumer as opposed to listener, then there's no need for that particular event funnelling and those lines here can become one simple `for` loop.